mem0/vector_stores/azure_ai_search.py

```diff
@@ -0,0 +1,231 @@
+
+try:
+    from azure.core.credentials import AzureKeyCredential
+    from azure.core.exceptions import ResourceNotFoundError
+    from azure.search.documents import SearchClient
+    from azure.search.documents.indexes import SearchIndexClient
+    from azure.search.documents.indexes.models import (
+        HnswAlgorithmConfiguration,
+        ScalarQuantizationCompression,
+        SearchField,
+        SearchFieldDataType,
+        SearchIndex,
+        SimpleField,
+        VectorSearch,
+        VectorSearchProfile,
+    )
+    from azure.search.documents.models import VectorizedQuery 
+except ImportError:
+    raise ImportError("The 'azure-search-documents' library is required. Please install it using 'pip install azure-search-documents==11.5.1'.")
+
+ 
+
+ 
+   
+logger = logging.getLogger(__name__)  
+  
+class OutputData(BaseModel):  
+    id: Optional[str]  
+    score: Optional[float]  
+    payload: Optional[dict]  
+  
+class AzureAISearch(VectorStoreBase):  
+    def __init__(self, service_name, collection_name, api_key, embedding_model_dims, use_compression):  
+        """Initialize the Azure Cognitive Search vector store.  
+  
+        Args:  
+            service_name (str): Azure Cognitive Search service name.  
+            collection_name (str): Index name.  
+            api_key (str): API key for the Azure Cognitive Search service.  
+            embedding_model_dims (int): Dimension of the embedding vector.  
+        """  
+        self.index_name = collection_name  
+        self.collection_name = collection_name
+        self.embedding_model_dims = embedding_model_dims  
+        self.use_compression = use_compression
+        self.search_client = SearchClient(endpoint=f"https://{service_name}.search.windows.net",  
+                                          index_name=self.index_name,  
+                                          credential=AzureKeyCredential(api_key))  
+        self.index_client = SearchIndexClient(endpoint=f"https://{service_name}.search.windows.net",  
+                                              credential=AzureKeyCredential(api_key))  
+        self.create_col()  #create the collection / index
+  
+    def create_col(self):  
+        """Create a new index in Azure Cognitive Search."""  
+        vector_dimensions = self.embedding_model_dims  # Set this to the number of dimensions in your vector 
+
+        
+        if self.use_compression:
+            vector_type = "Collection(Edm.Half)"
+        else:
+            vector_type = "Collection(Edm.Single)"
+
+        fields = [  
+            SimpleField(name="id", type=SearchFieldDataType.String, key=True),  
+            SearchField(name="vector", type=vector_type,
+                searchable=True, vector_search_dimensions=vector_dimensions, vector_search_profile_name="my-vector-config"),
+            SimpleField(name="payload", type=SearchFieldDataType.String, searchable=True)  
+        ]  
+
+
+        if self.use_compression:
```

Comment:
Can you please make line 78 and line 65 into one `if` block. It will more efficient